activesupport/test/message_encryptor_test.rb

```diff
@@ -132,3 +139,111 @@ def munge(base64_string)
       ::Base64.strict_encode64(bits)
     end
 end
+
+module SharedMessageMetadataTests
+  def setup
+    @secret    = SecureRandom.random_bytes(32)
+    @encryptor = ActiveSupport::MessageEncryptor.new(@secret, encryptor_options)
+
+    super
+  end
+
+  def test_encryption_and_decryption_with_same_purpose
+    message = "a security question's answer"
+    encrypted_message = @encryptor.encrypt_and_sign(message, purpose: "login")
+
+    assert_equal message, @encryptor.decrypt_and_verify(encrypted_message, purpose: "login")
+
+    encrypted_message = @encryptor.encrypt_and_sign(message)
+
+    assert_equal message, @encryptor.decrypt_and_verify(encrypted_message)
```

Comment:
Slight addendum to the newlines between assertions from yesterday, when there's 2 or more cases like this, it's usually easier to group the act and assertions together:

```ruby
message = "a security question's answer"

login = @encryptor.encrypt_and_sign(message, purpose: "login")
assert_equal message, @encryptor.decrypt_and_verify(login, purpose: "login")

no_purpose = @encryptor.encrypt_and_sign(message)
assert_equal message, @encryptor.decrypt_and_verify(no_purpose)
```

(It can also be nice not to reuse the variable name :))